plugins/modules/mso_schema_site_external_epg.py

```diff
@@ -166,20 +192,25 @@ def main():
     l3out = module.params.get("l3out")
     l3out_schema = module.params.get("l3out_schema")
     l3out_template = module.params.get("l3out_template")
+    l3out_template_id = module.params.get("l3out_template_id")
     l3out_on_apic = module.params.get("l3out_on_apic")
     route_reachability = module.params.get("route_reachability")
+    l3out_uuid = module.params.get("l3out_uuid")
+    l3out_on_l3out_template = module.params.get("l3out_on_l3out_template")
     state = module.params.get("state")
 
     mso = MSOModule(module)
 
-    l3out_template = template if l3out_template is None else l3out_template.replace(" ", "")
-    l3out_schema = schema if l3out_schema is None else l3out_schema
-    l3out_schema_id = mso.lookup_schema(l3out_schema)
+    if not l3out_on_l3out_template:
+        l3out_template = template if l3out_template is None else l3out_template.replace(" ", "")
+        l3out_schema = schema if l3out_schema is None else l3out_schema
+        l3out_schema_id = mso.lookup_schema(l3out_schema)
 
     mso_schema = MSOSchema(mso, schema, template, site)
     mso_objects = mso_schema.schema_objects
 
     mso_schema.set_template_external_epg(external_epg, fail_module=False)
+    mso_template = MSOTemplate(mso, "l3out", l3out_template, l3out_template_id)
```

Comment:
do we need to create this for scenarios where it is not l3out_on_l3out_template?